packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { UsageError } from "@fluidframework/driver-utils";
+import { MonitoringContext } from "@fluidframework/telemetry-utils";
+import { IGCRuntimeOptions } from "..";
+import { IContainerRuntimeMetadata } from "../summary";
+import {
+	defaultInactiveTimeoutMs,
+	defaultSessionExpiryDurationMs,
+	disableTombstoneKey,
+	gcTestModeKey,
+	IGarbageCollectorConfigs,
+	oneDayMs,
+	runGCKey,
+	runSessionExpiryKey,
+	runSweepKey,
+	trackGCStateKey,
+} from "./gcDefinitions";
+
+/**
+ * Generates configurations for the Garbage Collector that it uses to determine what to run and how.
+ * @param gcOptions - The garbage collector runtime options.
+ * @param metadata - The container runtime's metadata.
+ * @param baseSnapshotGCVersion - The GC version of the base snapshot.
+ * @param existing - Whether the container is new or an existing one.
+ * @param mc - The monitoring context for reading configs from the config provider.
+ * @returns The garbage collector configurations.
+ */
+export function generateGCConfigs(
+	gcOptions: IGCRuntimeOptions,
+	metadata: IContainerRuntimeMetadata | undefined,
+	baseSnapshotGCVersion: number | undefined,
+	existing: boolean,
+	mc: MonitoringContext,
+): IGarbageCollectorConfigs {
+	const configs: IGarbageCollectorConfigs = {
+		gcEnabled: false,
+		sweepEnabled: false,
+		shouldRunGC: false,
+		shouldRunSweep: false,
+		testMode: false,
+		tombstoneMode: false,
+		trackGCState: false,
+		sessionExpiryTimeoutMs: undefined,
+		sweepTimeoutMs: undefined,
+		inactiveTimeoutMs: 0,
+	};
+
+	/**
+	 * Sweep timeout is the time after which unreferenced content can be swept.
+	 * Sweep timeout = session expiry timeout + snapshot cache expiry timeout + one day buffer.
+	 *
+	 * The snapshot cache expiry timeout cannot be known precisely but the upper bound is 5 days.
+	 * The buffer is added to account for any clock skew or other edge cases.
+	 * We use server timestamps throughout so the skew should be minimal but make it 1 day to be safe.
+	 */
+	const computeSweepTimeout = (sessionExpiryTimeoutMs: number | undefined) => {
```

Comment:
Now that this code is in a separate file, this could be a non-exported function at the top level

Reply:
done.